### PR TITLE
Add support for MultiAddress

### PR DESCRIPTION
**Pull request by vgeddes**
[MultiAddress](https://crates.parity.io/sp_runtime/enum.MultiAddress.html) is the new default address type in Substrate master (See https://github.com/paritytech/substrate/pull/7380).

Tested against the latest `node-template` from Substrate master using the following:
```bash
go test -v -timeout 30s -run ^TestChain_SubmitExtrinsic$ github.com/centrifuge/go-substrate-rpc-client/v2/teste2e
```

Leaving this PR in draft mode as a bunch of tests still need to be updated 🙂

### CODE REVIEW

File: types/extrinsic_signature.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ type ExtrinsicSignatureV3 struct {
 }
 
 type ExtrinsicSignatureV4 struct {
```
Comment (vgeddes) [inline types/extrinsic_signature.go:27]: For consistency with polkadot-js, I've not added a `ExtrinsicSignatureV5`.

File: types/extrinsic_signature.go [hunk 2]
```diff
@@ -25,7 +25,7 @@ type ExtrinsicSignatureV3 struct {
...
-	Signer    Address
```
Comment (vedhavyas) [inline types/extrinsic_signature.go:28]: this will break for all the chains without the multiaddress.

Comment (vgeddes) [reply types/extrinsic_signature.go:28]: Good point, there are a couple of chains also using substrate master who for reasons of simplicity prefer not to use MultiAddress.

We could this solve this by making `ExtrinsicSignatureV4.Signer` an interface type:

```
type ExtrinsicSignatureV4 {
    Signer interface{}
    ...
}
```

Comment (vedhavyas) [reply types/extrinsic_signature.go:28]: I think we could abstract out the Signerwith Address and MultiAddress as default implementations. This wouldn't break the compatibility

**Comment by Gauthamastro** [comment]
Guys, we are working on ChainBridge, and this PR needs to be merged since this issue a blocker for us. Can you please merge the PR if it is ready? 
@vedhavyas @vgeddes

**Comment by vgeddes** [comment]
> Guys, we are working on ChainBridge, and this PR needs to be merged since this issue a blocker for us. Can you please merge the PR if it is ready?
> @vedhavyas @vgeddes

Hey @Gauthamastro we're kinda using our own fork of GSRPC now as MultiAddress support is only one of about a half-dozen changes required to support Substrate 3.0 fully (at least for our specific needs):

https://github.com/Snowfork/go-substrate-rpc-client

We don't have an ETA for merging this work upstream, as it is lower priority for us right now, given our schedule.

**Comment by Gauthamastro** [comment]
> > Guys, we are working on ChainBridge, and this PR needs to be merged since this issue a blocker for us. Can you please merge the PR if it is ready?
> > @vedhavyas @vgeddes
> 
> Hey @Gauthamastro we're kinda using our own fork of GSRPC now as MultiAddress support is only one of about a half-dozen changes required to support Substrate 3.0 fully (at least for our specific needs):
> 
> https://github.com/Snowfork/go-substrate-rpc-client
> 
> We don't have an ETA for merging this work upstream, as it is lower priority for us right now, given our schedule.

Ohh okay thanks for the fork. We will use that

**Comment by vedhavyas** [comment]
@vgeddes we started working on v3 changes. I have pulled your work on Multi address from this PR. Closing this in favor of that(#143).

**Comment by vgeddes** [comment]
@vedhavyas cool cool. There's a bunch of other V3 changes in our fork if you want to pick up those too: https://github.com/Snowfork/go-substrate-rpc-client. They also include RPC APIs for offchain storage.

**Pull request closed without merging**